app/controllers/routes_controller.rb

```diff
@@ -0,0 +1,28 @@
+class RoutesController < ApplicationController
+  helper_method :sort_column, :sort_direction
+
+  def index
+    @routes = Route.joins(:user).where(nil)
+    @routes = @routes.with_full_text_search(params[:search]) if params[:search].present?
+    @routes = @routes.order(sort_column + " " + sort_direction)
+  end
+
+  def show
+    @route = Route.find(params[:id])
+  end
+
+  def new
+  end
+
+  private
+
+  def sort_column
+    whitelist = %w(name users.fname location tape_color
```

Comment:
Have to manually specify whitelist because `Route.column_names` does not include `users.fname` which is necessary for searching on the setter's name.